src/users/users.service.ts

```diff
@@ -1,36 +1,59 @@
 import { Injectable } from '@nestjs/common';
 import { CreateUserDto } from './dto/create-user.dto';
-import { SetSshDto } from './dto/set-ssh.dto';
+import { SetSshDto } from 'src/users/dto/set-ssh.dto';
 import { PrismaService } from 'src/prisma.service';
 import { User, Prisma } from '@prisma/client';
 import { SanitizedUser } from './types/sanitized-user.type';
 import { exclude } from 'src/utils/prisma-exclude';
+import * as bcrypt from 'bcrypt';
 
 @Injectable()
 export class UsersService {
-  constructor(private prisma: PrismaService) {}
-
-  private passwd_encrypt(password: string) {
-    // FIXME: implement encrypting passwords
-    return password;
-  }
+  constructor(private prisma: PrismaService) { }
 
   private sanitizeOutput(user: User): SanitizedUser {
     return exclude(user, ['id', 'email_nonce', 'password', 'createdAt', 'updatedAt']);
   }
 
-  async create(user: CreateUserDto): Promise<SanitizedUser> {
+  private async passwd_encrypt(password: string): Promise<string> {
+    return await bcrypt.hash(password, 10);
+  }
+
+  async create(user: CreateUserDto, uuid: string): Promise<SanitizedUser> {
     return this.sanitizeOutput(await this.prisma.user.create({
       data: {
         username: user.username,
         email: user.email,
-        password: this.passwd_encrypt(user.password),
+        password: await this.passwd_encrypt(user.password),
         isAdmin: false,
+        email_nonce: uuid
       }
     }));
   }
 
-  async findOne(userUniqueInput: Prisma.UserWhereUniqueInput): Promise<SanitizedUser|null> {
+  async validateEmail(email: string): Promise<boolean> {
+    // Check that email address doesn't already exist in the db
+    return !await this.findOne({ email });
+  }
+
+  async validateEmailNonce(uuid: string): Promise<boolean> {
+    await this.prisma.user.update({
```

Comment:
I suggest using a try-catch block here, since we don't won't to return true if user update fails for some reason.

Reply:
try catch or more something like 
```js
let user = await this.prisma.user.update(); 
if (!user) {
return false
}
return true;
```

Reply:
I'm not sure if prisma will just return null on error. If it throws an uncaught exception, it may crush our API. I found this code example in documentation: https://www.prisma.io/docs/concepts/components/prisma-client/handling-exceptions-and-errors